cmd/habitatctl/commands/start.go

```diff
@@ -22,15 +22,44 @@ func parseFlags(args []string) ([]string, []string) {
 	return flags, nonflags
 }
 
+var procName string
+
+var ipfsCmd = &cobra.Command{
```

Comment:
Is the reason we're making a specific IPFS subcommand here just so that we have some code to intercept and process flags before passing it to the script?

Reply:
yeah, the main reason is that we want some flags that are required for starting ipfs but not necessarily for other process in general. (see the env variable IPFS_PATH). i think ultimately we want to be able to pass env variables through from here, and there's not a good way to do taht without having some command-specific code (you either have to do it here or deeper in the code get the value from some flag, since cobra doesn't support something like that). so the hack is to require a certain flag to be set for the ipfs command and then set the env variable when sending the request in line 46 of commands/start.go in the final commit (sorry the commits are super messy for this since i was playing around with different ways to do it with cobra). i think at some point we may wanna not use cobra since we want to pass so many things down to the process level and cobra is not really meant for that.

Reply:
What if we pass them in the same way we do app specific flags, after the --? So for example we might run `habitatctl start ipfs -- IPFS_PATH=abcd --flag flagValue positional_value1 positional_value2`. Your parsing function could be changed to handle env vars, flags and positional args.